### PR TITLE
Add `GPS receiver` device to provide `senseloc` capability

### PR DESCRIPTION
**Pull request by byorgey**
Add a new `GPS receiver` device which enables the `senseloc` capability (and hence the `whereami` command).  Towards #26 .

- My immediate motivation is that I want to be able to define `excursion : cmd unit -> cmd unit` which executes the given command and then returns to the same location and orientation as before.  Being able to use the `whereami` command is one of the last missing pieces of machinery necessary to be able to do this in classic mode (the other is a `heading` command, see #955).
- The proposed recipe for `GPS receiver` is `antenna + circuit + clock + compass`, which is a somewhat difficult recipe (`antenna` requires `silver` which requires a `deep mine`; a `clock` requires `quartz` + a bunch of `iron gears`, etc.).  One might wonder whether we should make the recipe easier since finding out where you are seems like a kind of fundamental operation.  However, consider that in order to even be able to make use of the result of `whereami` you need at least (1) an `ADT calculator` to deal with the pair (which transitively requires `typewriter` -> `circuit` -> `silicon` -> `quartz`) (2) probably things like `comparator` and `calculator` to do anything useful with the coordinates.  By the time you have those things you can definitely already build `circuit` + `clock` + `compass`, and you're probably not that far away from getting some `silver` for an `antenna`.  Also, in practice it's an interesting/fun constraint to build up machinery that has to work entirely based on *relative* position without being able to find out your absolute coordinates.
- For some reason this is causing `Testing/508-capability-subset` to fail.  I think perhaps it is due to #397 ?  I will investigate. *EDIT*: unfortunately, that wasn't it!

### CODE REVIEW

**Comment by xsebek** [comment]
What is going on? 😄 
```
Testing/508-capability-subset:                                                                       FAIL (0.28s)
[374](https://github.com/swarm-game/swarm/actions/runs/3839118001/jobs/6536452772#step:22:375)
        test/integration/Main.hs:263:
[375](https://github.com/swarm-game/swarm/actions/runs/3839118001/jobs/6536452772#step:22:376)
        Player exception: Fatal error: BASE: I could not build my robot! What happened?!
[376](https://github.com/swarm-game/swarm/actions/runs/3839118001/jobs/6536452772#step:22:377)
        
[377](https://github.com/swarm-game/swarm/actions/runs/3839118001/jobs/6536452772#step:22:378)
        Use -p '/Regression tests.Testing\/508-capability-subset/' to rerun this test only.

**Comment by byorgey** [comment]
Unfortunately fixing #397 didn't resolve the failing test. The `build` is failing with an error saying "You do not have required devices, please obtain: GPS receiver", but I don't understand why.  The `GPS` and `Tardis` entities defined in the scenario both provide the `senseloc` capability so it shouldn't matter that we don't have a `GPS receiver`.  But it's like it's not seeing those.

**Comment by byorgey** [comment]
doh!

https://github.com/swarm-game/swarm/blob/629cff3b703541fac332b83973dcc9c4bf8a2fc3/src/Swarm/Game/Entity.hs#L295-L296

Hint: combining two `Map Capability [Entity]` values with `<>` doesn't exactly do what we want. :facepalm:

**Comment by byorgey** [comment]
Well, finally got the tests to pass. Even though #397 wasn't the cause I'm glad I thought it was, because it gave me the push to finally get that sorted. :grin:

File: data/scenarios/Testing/956-GPS.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+version: 1
+name: Test GPS receiver device
+description: |
+  Test the `whereami` command and GPS receiver device.
+  https://github.com/swarm-game/swarm/issues/956
+objectives:
```
Comment (kostmo) [inline data/scenarios/Testing/956-GPS.yaml:6]: Add a `goal` here that explains what the solution is doing to meet the `condition`

**Review by kostmo — APPROVED** [review]

File: data/scenarios/Testing/956-GPS.yaml (hunk 1 above)
Comment (byorgey) [reply data/scenarios/Testing/956-GPS.yaml:6]: Oh, good idea.  While I was at it I also thought of a way to make the test much better --- what do you think now?

Comment (kostmo) [reply data/scenarios/Testing/956-GPS.yaml:6]: Yes, using `random` is great.

**Review by kostmo — APPROVED** [review]

**Pull request merged**